### PR TITLE
chore: update `pre-commit`

### PR DESCRIPTION
**Pull request by Stranger6667**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -38,7 +38,7 @@ dependencies = [
     "httpx>=0.22.0,<1.0",
     "hypothesis>=6.31.6,<7",
     "hypothesis_graphql>=0.9.0,<1",
-    "hypothesis_jsonschema>=0.22.1,<1",
+    "hypothesis_jsonschema>=0.22.1,<0.23",
```
Comment (allanlewis) [inline pyproject.toml:41]: **question:** What's the reason for this change? I use `hypothesis-jsonschema` explicitly in my code so I've had to restrict it to `<0.23` for the time being.

Comment (Stranger6667) [reply pyproject.toml:41]: I guess that I expected that the changes in `hypothesis-jsonschema` are not compatible with Schemathesis for now. Need to check it

Comment (allanlewis) [reply pyproject.toml:41]: OK, that makes sense. It's just a bit confusing given that previously this was much looser, and I didn't see anything in the changelog.

Perhaps it's worth pushing the maintainers of `hypothesis-jsonschema` to release a 1.0 and follow SemVer from there 🤔

Comment (Stranger6667) [reply pyproject.toml:41]: We plan to overhaul the internals there soon (to finally solve the recursive reference issue), so I hope it will happen in the foreseeable future

Comment (allanlewis) [reply pyproject.toml:41]: Great, thanks!
